Tasks/NuGetInstaller/NuGetInstaller.ps1

```diff
@@ -62,15 +65,40 @@ if(!$nuGetPath)
     $nuGetPath = Get-ToolPath -Name 'NuGet.exe';
 }
 
+if (-not $nugetPath)
+{
+    throw (Get-LocalizedString -Key "Unable to locate {0}" -ArgumentList 'nuget.exe')
+}
+
 if($nuGetRestoreArgs)
 {
+    if($nuGetRestoreArgs.ToLowerInvariant().Contains("-configfile"))
+    {
+        Write-Warning "ConfigFile was passed as a command line parameter, which may be ignored in certain parts of this task. Please specify the config file in the build definition instead."
+    }
+
     $args = ($args + " " + $nuGetRestoreArgs);
 }
 
-
-if (-not $nugetPath)
+if($nugetConfigPath -ne $env:Build_SourcesDirectory)
 {
-    throw (Get-LocalizedString -Key "Unable to locate {0}" -ArgumentList 'nuget.exe')
+    $args = $args + " -configfile " + $tempNuGetConfigPath
```

Comment:
It looks like $tempNuGetConfigPath could contain spaces and isn't already encapsulated in double quotes. Looks like this should be:
$args = "$args -configfile `"$tempNuGetConfigPath`""

This is building up the args for nuget.exe right? not for a ps cmdlet?